src/swap/components/SwapProvider.tsx

```diff
@@ -184,21 +184,74 @@ export function SwapProvider({
           return handleError({ swapError: response });
         }
 
-        onSubmit?.(response);
+        const { transaction, approveTransaction } = response;
+
+        if (approveTransaction?.data) {
```

Comment:
This condition has "a lot" of code.

Same for the "catch" part. How all this part can be exported into an utility fuction, and have stronger unit-test around that function?